src/arithmetic.jl

```diff
@@ -0,0 +1,501 @@
+# This file is part of the TaylorSeries.jl Julia package, MIT license
+#
+# Luis Benet & David P. Sanders
+# UNAM
+#
+# MIT Expat license
+#
+
+# Arithmetic operations: +, -, *, /
+
+## Equality ##
+for T in (:Taylor1, :TaylorN)
+    @eval begin
+        =={T<:Number,S<:Number}(a::$T{T}, b::$T{S}) = ==(promote(a,b)...)
+        function =={T<:Number}(a::$T{T}, b::$T{T})
+            a, b = fixorder(a, b)
+            @simd for i in eachindex(a.coeffs)
+                @inbounds (a.coeffs[i] == b.coeffs[i]) || return false
+            end
+            true
+        end
+    end
+end
+
+==(a::HomogeneousPolynomial, b::HomogeneousPolynomial) = a.coeffs == b.coeffs
+
+
+function iszero{T}(a::HomogeneousPolynomial{T})
+    @simd for i in eachindex(a.coeffs)
+        @inbounds (a.coeffs[i] == zero(T)) || return false
+    end
+    return true
+end
+
+
+## zero and one ##
+zero{T<:Number}(a::Taylor1{T}) = Taylor1(zero(a.coeffs[1]), a.order)
+one{T<:Number}(a::Taylor1{T}) = Taylor1(one(a.coeffs[1]), a.order)
```

Comment:
```
julia v0.5> x, y = set_variables("x y")
2-element Array{TaylorSeries.TaylorN{Float64},1}:
  1.0 x + 𝒪(‖x‖⁷)
  1.0 y + 𝒪(‖x‖⁷)

julia v0.5> t = Taylor1(x, 10)
  1.0 x + 𝒪(‖x‖⁷) + 𝒪(t¹¹)

julia v0.5> one(eltype(t))
 1.0 + 𝒪(‖x‖¹)
```
`one(a.coeffs[1])` is the same as `one(typeof(a.coeffs[1]))` so my suggestion should always work.